### PR TITLE
Python 3.11 and new emscripten-wasm32 platform

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

**Comment by martinRenou** [comment]
For some reason, the RTD mamba environment creation fails on this PR, although it does works on the main branch and the `build-environment.yml` file has not changed in this PR.

It fails with a memory issue (turns out I also had an issue with micromamba locally with memory piling up)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -11,11 +11,11 @@ defaults:
     shell: bash -l {0}
 
 jobs:
-  build:
+  lint:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v2
```
Comment (jtpio) [inline .github/workflows/build.yml:18]: This downgrade to `v2` is probably unwanted?

Comment (martinRenou) [reply .github/workflows/build.yml:18]: Thanks for the review, indeed that's a rebase issue

**Comment by martinRenou** [comment]
Let's merge (probably make an alpha release) and [test this](https://github.com/jupyterlite/xeus-python-kernel/pull/167) more properly before the final

**Comment by martinRenou** [comment]
meeseeksdev backport to 0.x

**Pull request merged**